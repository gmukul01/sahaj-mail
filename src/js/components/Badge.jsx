import React from 'react';
import PropTypes from 'prop-types';

const Badge = props => {
	const { className, badgeContent, children } = props;

	return (
		<button className={className}>
			{children}
			{badgeContent && badgeContent !== 0 ? <span className="badge">{badgeContent}</span> : null}
		</button>
	);
};

Badge.propTypes = {
	className: PropTypes.string,
	badgeContent: PropTypes.number,
	children: PropTypes.oneOfType([PropTypes.node, PropTypes.string])
};

export default React.memo(Badge);
