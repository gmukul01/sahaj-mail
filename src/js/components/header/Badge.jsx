import React from 'react';

const Badge = props => {
	const { className, badgeContent, children } = props;

	return (
		<button className={className}>
			{children}
			<span className="badge">{badgeContent}</span>
		</button>
	);
};

export default React.memo(Badge);
