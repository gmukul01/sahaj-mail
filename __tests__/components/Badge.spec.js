import React from 'react';
import renderer from 'react-test-renderer';

import Badge from 'components/Badge';

describe('Badge Component', () => {
	it('should match snapshot when badgeContent is non zero', () => {
		const badge = renderer
			.create(
				<Badge className="badge-email" badgeContent={4}>
					<i className={`fas fa-bell`} />
				</Badge>
			)
			.toJSON();
		expect(badge).toMatchSnapshot();
	});

	it('should match snapshot when badgeContent is zero', () => {
		const badge = renderer
			.create(
				<Badge className="badge-email" badgeContent={0}>
					<i className={`fas fa-bell`} />
				</Badge>
			)
			.toJSON();
		expect(badge).toMatchSnapshot();
	});
});
