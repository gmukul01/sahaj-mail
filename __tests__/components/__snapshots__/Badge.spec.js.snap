// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component should match snapshot when badgeContent is non zero 1`] = `
<button
  className="badge-email"
>
  <i
    className="fas fa-bell"
  />
  <span
    className="badge"
  >
    4
  </span>
</button>
`;

exports[`Badge Component should match snapshot when badgeContent is zero 1`] = `
<button
  className="badge-email"
>
  <i
    className="fas fa-bell"
  />
</button>
`;
